src/routers/SwitchRouter.js

```diff
@@ -177,8 +196,16 @@ export default (routeConfigs, config = {}) => {
       const isBackEligible =
         action.key == null || action.key === activeChildLastState.key;
       if (action.type === NavigationActions.BACK) {
-        if (isBackEligible && shouldBackNavigateToInitialRoute) {
-          activeChildIndex = initialRouteIndex;
+        if (isBackEligible && backBehavior !== 'none') {
```

Comment:
these ifs can be flattened, and IMO it is easier to read that way. Redundant `else return state` can be eliminated.